packages/block-editor/src/components/block-bindings-button/index.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * WordPress dependencies
+ */
+import { ToolbarButton } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { connection } from '@wordpress/icons';
+import { _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+
+export default function BlockBindingsButton( { clientId } ) {
+	const { blockAttributes } = useSelect(
+		( select ) => {
+			const { getBlockAttributes } = select( blockEditorStore );
+			return {
+				blockAttributes: getBlockAttributes( clientId ),
+			};
+		},
+		[ clientId ]
+	);
```

Comment:
Suggestion: It's usually better to derive boolean values inside the `mapSelect`, so the editor doesn't rerender component everytime the attributes change.

Example:

```jsx
export default function BlockBindingsButton( { clientId } ) {
	const isConnected = useSelect(
		( select ) => {
			const attributes =
				select( blockEditorStore ).getBlockAttributes( clientId );

			return !! attributes?.metadata?.bindings;
		},
		[ clientId ]
	);

	return isConnected ? (
		<ToolbarButton
			icon={ connection }
			label={ _x(
				// TODO: Let's get this naming right
				'Connected to a block bindings source',
				'block toolbar button label'
			) }
			iconSize={ 24 }
		/>
	) : null;
}
```

Reply:
Totally 👍 585d5a3ef55272bb370d90b6dfd747bdd34dace4